From 2b251d69351312d632e41188d5f9f7cbf394331a Mon Sep 17 00:00:00 2001
From: Ross Lagerwall <rosslagerwall@gmail.com>
Date: Fri, 11 Oct 2013 22:04:51 +0200
Subject: [PATCH] dav: Set file type to regular by default

Some servers send empty resourcetype nodes or don't send the node at all
for regular files (the spec says that it defaults to empty).  Set the
file type to regular by default.

https://bugzilla.gnome.org/show_bug.cgi?id=706798
---
 daemon/gvfsbackenddav.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/daemon/gvfsbackenddav.c b/daemon/gvfsbackenddav.c
index 2feb980..39995d6 100644
--- a/daemon/gvfsbackenddav.c
+++ b/daemon/gvfsbackenddav.c
@@ -991,7 +991,7 @@ ms_response_to_file_info (MsResponse *response,
   g_file_info_set_name (info, basename);
   g_file_info_set_edit_name (info, basename);
 
-  file_type = G_FILE_TYPE_UNKNOWN;
+  file_type = G_FILE_TYPE_REGULAR;
   mime_type = NULL;
 
   have_display_name = FALSE;
@@ -1013,7 +1013,6 @@ ms_response_to_file_info (MsResponse *response,
           if (node_has_name (node, "resourcetype"))
             {
               file_type = parse_resourcetype (node);
-              g_file_info_set_file_type (info, file_type);
             }
           else if (node_has_name (node, "displayname") && text)
             {
@@ -1059,6 +1058,7 @@ ms_response_to_file_info (MsResponse *response,
         }
     }
 
+  g_file_info_set_file_type (info, file_type);
   if (file_type == G_FILE_TYPE_DIRECTORY)
     {
       icon = g_themed_icon_new ("folder");
-- 
1.8.4.2

